stdlib/stdlib-external/ascutils/dataStructures.ts

```diff
@@ -68,6 +68,26 @@ export function allocateString(size: u32): u32 {
   return str
 }
 
+export function singleCharacterString(char: u8): u32 {
+  let s = allocateString(1)
+  store<u8>(s, char, 8)
+  return s | GRAIN_GENERIC_HEAP_TAG_TYPE
+}
+
+export function twoCharacterString(char1: u8, char2: u8): u32 {
+  let s = allocateString(2)
+  store<u8>(s, char1, 8)
+  store<u8>(s, char2, 8 + 1)
+  return s | GRAIN_GENERIC_HEAP_TAG_TYPE
+}
+
+export function ascStringToGrainString(str: string): u32 {
```

Comment:
Bad news. I did some research (and testing), and we can't use this at all. The way that static memory allocation works is all of the bytes are loaded into the beginning of the WebAssembly memory upon initialization. That's maybe fine if you're producing a single wasm file and offset where your dynamic memory begins, but multiple modules will each write over each other, not to mention that it trashes the heap. It's not apparent in any of the tests because `malloc` allocates from the tail end of the memory. So you'd randomly run into issues after you start using lots of memory 🙂 

(I tested this by putting a large static string in runtime.ts, looking at the start of the memory, and confirming that it was overrun with the bytes in my string.)

Good thing you've already put all of the character constants in the code!